pkg/sidecar/config.go

```diff
@@ -17,10 +17,29 @@ type Config struct {
 	// The path to the process to launch.
 	Cmd string
 
-	// The arguments of the process to launch.
+	// The arguments of the process to launch. Deprecated, use CmdArgsArray.
 	CmdArgs string
 
-	// Signal external process via PID file
+	// An argument vector to be passed to Cmd when invoking a command. If not empty,
+	// CmdArgs is ignored.
+	CmdArgsArray []string
+
+	// Should stdin be attached when running a sub-command? Historically spiffe-helper did not attach stdin,
+	// but generally when running as a wrapper process, stdin should be attached.
+	CmdAttachStdin bool
```

Comment:
This seems like it should just be the standard behavior.

Reply:
I'd usually agree, but I'm hesitant about what it might break. I've run into some exciting behaviour with shell scripts like

```
while read -r FOO
do
    ... something here ...
done < <(jq ....)
```

where surprising and exciting things happen if `...something here...` changes to consume from `stdin`.

It's probably not a realistic concern though, and rather than adding a config option I think you're right it should be the default. It can just be called out in the release notes, with a note that anyone who doesn't want stdin connected can close stdin before forking `spiffe-helper`, e.g. `spiffe-helper ... <&-` from the shell.

I'll update when I pick this back up.

Reply:
PR updated, and cut down to just the minimum change